compiler/rustc_lint/src/tail_expr_drop_order.rs

```diff
@@ -21,8 +21,7 @@ declare_lint! {
     /// Your discretion on the new drop order introduced by Edition 2024 is required.
     ///
     /// ### Example
-    /// ```rust,edition2024
-    /// #![feature(shorter_tail_lifetimes)]
+    /// ```rust,edition2021
```

Comment:
existing, but "The `tail_expr_drop_order` lint looks for those values generated at the tail expression location, that of type with a significant `Drop` implementation, such as locks." seems wrong to me.

```rust
    /// The `tail_expr_drop_order` lint looks for those values generated at the tail expression location,
    /// whose type has a significant `Drop` implementation, such as locks.
```

Reply:
I have tweaked the wording for clarity